codice/ddf
# DDF-3855: Add CSRF Protections to Jolokia and Intrigue Endpoints

#### What does this PR do?
Adds the header "X-Requested-With" as a required header on the backend so that all incoming requests to anything under `/admin/jolokia` or `/search/catalog/internal` must contain this header or a 403 will be returned. Value of this header can be anything, only it's presence is checked. The incoming request must also have either an origin or referer header that matches the target origin or a 403 will be returned. Also adds this header to any necessary UI packages.

#### Who is reviewing it? 
@djblue 
@rzwiefel 
@tbatie 

#### Select relevant component teams: 
@codice/security 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler 
@stustison

#### How should this be tested?
Make sure the admin UI and Intrigue still work. Watch network requests and make sure no 403s are returned. Try accessing jolokia & intrigue internal URLs directly and ensure that they cannot be accessed - a 403 should be returned. See below for a list of packages and contexts that make jolokia and intrigue contexts that should be checked for compatibility.

#### Any background context you want to provide?
Currently holding out on merging this PR while downstream Codice project frontends are checked and modified to ensure compatibility with this new requirement.

The following UI packages under `/ui/packages` make jolokia or intrigue calls and were checked for functionality to ensure that the header was being attached to any outgoing jolokia requests:
`jQuery-updated` ui - /admin
`ok` spatial-geocoding-admin-module - /admin/spatial-config
`ok` resourcemanagement-usage-ui - /admin/data-usage
`ok` resourcemanagement-querymonitor-ui - /admin/queries
`ok` registry-admin-remote-ui - /admin/registry/remote
`ok` registry-admin-local-ui - /admin/registry/local
`ok` catalog-admin-module-sources - /admin/sources
`fetch-updated` catalog-admin-module-maplayers - /admin/map-layers
`fetch-updated` catalog-admin-module-layout - /admin/default-layout
`fetch-updated` broker-undelivered-messages-ui - /admin/undeliveredMessages
`ok` admin-security-certificate-ui - /admin/certificate
`used in test-only` admin-metrics-ui - /admin/metrics
`fetch-updated` admin-core-logviewer - /admin/logviewer

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-3855

#### Screenshots (if appropriate)
#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
